owncloudApp/src/main/java/com/owncloud/android/ui/activity/BaseActivity.java

```diff
@@ -274,4 +278,53 @@ public void hideSoftKeyboard() {
             );
         }
     }
-}
\ No newline at end of file
+
+    /**
+     * Show loading dialog
+     */
+    public void showLoadingDialog(int messageId) {
+        // grant that only one waiting dialog is shown
+        dismissLoadingDialog();
+        // Construct dialog
+        Fragment frag = getSupportFragmentManager().findFragmentByTag(DIALOG_WAIT_TAG);
+        if (frag == null) {
+            Log_OC.d(TAG, "show loading dialog");
+            LoadingDialog loading = LoadingDialog.newInstance(messageId, false);
+            FragmentManager fm = getSupportFragmentManager();
+            FragmentTransaction ft = fm.beginTransaction();
+            loading.show(ft, DIALOG_WAIT_TAG);
+            fm.executePendingTransactions();
+        }
+    }
+
+    /**
+     * Dismiss loading dialog
+     */
+    public void dismissLoadingDialog() {
+        Fragment frag = getSupportFragmentManager().findFragmentByTag(DIALOG_WAIT_TAG);
```

Comment:
can you refactor to this?

```
Fragment frag = getSupportFragmentManager().findFragmentByTag(DIALOG_WAIT_TAG);

if (frag == null) return;

Log_OC.d(TAG, "dismiss loading dialog");
LoadingDialog loading = (LoadingDialog) frag;
loading.dismiss();
```

Reply:
if this file was in kotlin, the refactor would be:

val frag = getSupportFragmentManager().findFragmentByTag(DIALOG_WAIT_TAG) as LoadingDialog ?: return

Log_OC.d(TAG, "dismiss loading dialog");
frag.dismiss();

Reply:
done